scala/scala-impl/src/org/jetbrains/plugins/scala/compiler/ScalaHydraCompilerConfigurationPanel.scala

```diff
@@ -71,15 +71,26 @@ class ScalaHydraCompilerConfigurationPanel(project: Project, settings: HydraComp
 
   def setHydraStoreDirectory(path: String): Unit = hydraStoreDirectoryField.setText(path)
 
+  val compilerRegex = """.*scala-compiler-(\d+\.\d+\.\d+)\.jar""".r
+
   def onDownload(): Unit = {
+
+    // we can't use `module.sdk.compilerVersion` because it assumes the *name* of the Sdk library
+    // matches `SDK-2.12.3` or the like. For the Scala project this is simply called `starr`, so we
+    // need to look inside and retrieve the actual classpath entries
     val scalaVersions = for {
       module <- project.scalaModules
-      scalaVersion <- module.sdk.compilerVersion
-    } yield scalaVersion
-
-    downloadVersionWithProgress(scalaVersions, selectedVersion)
-    settings.hydraVersion = selectedVersion
-    EditorNotifications.updateAll()
+      classpathFile <- module.sdk.compilerClasspath
+      mtch <- compilerRegex.findFirstMatchIn(classpathFile.getName)
+    } yield mtch.group(1)
+
+    if (scalaVersions.isEmpty)
+      Messages.showErrorDialog("Could not determine Scala version in this project.", "Hydra Plugin Error")
```

Comment:
I'm thinking we should log the classpath otherwise it will be harder to know if the error was produced because the scala-compiler jar was not there or maybe because the regular expression didn't work as expected.